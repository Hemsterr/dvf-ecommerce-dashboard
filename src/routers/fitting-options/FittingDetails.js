// @flow
import React from 'react'
import BookingItem from '../../components/BookingItem'
import FittingWorks from '../../components/FittingWorks'
import { fittingOptions, fittingSteps } from '../../constants/fittingOptions'

type Props = {
  fittingOption: string,
  handleSelectOption: Function,
}

class FittingDetails extends React.PureComponent<Props> {
  static defaultProps = {
    fittingOption: 'fitting-kit',
    handleSelectOption: () => {},
  }

  render() {
    const { fittingOption, handleSelectOption } = this.props
    const fitting = fittingOptions.find(item => item.type === fittingOption)
    const options = fittingOptions.filter(item => item.type !== fittingOption)

    return (
      <div className="fitting-options__fitting-details">
        <BookingItem description={fitting.description} type={fittingOption} />
        <FittingWorks
          options={fittingSteps[fittingOption]}
          title={fitting.title}
        />
        <div>
          {options.map(item => (
            <BookingItem
              handleSelectOption={handleSelectOption}
              type={item.type}
              key={item.type}
              isSmall
            />
          ))}
        </div>
      </div>
    )
  }
}

export default FittingDetails
