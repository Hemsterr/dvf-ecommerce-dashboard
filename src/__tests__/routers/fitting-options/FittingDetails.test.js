// Libs
import React from 'react'
import { render } from '@testing-library/react'

// Component
import FittingDetails from '../../../routers/fitting-options/FittingDetails'

const wrapper = (props = {}) => render(<FittingDetails {...props} />)

describe('FittingDetails component', () => {
  test('Render the FittingDetails component with default props', () => {
    const wrapperWithoutProps = wrapper()
    expect(wrapperWithoutProps.container).toMatchSnapshot()
    expect(FittingDetails.defaultProps.fittingOption).toEqual('fitting-kit')
    expect(FittingDetails.defaultProps.handleSelectOption()).toBeUndefined()
  })
})
