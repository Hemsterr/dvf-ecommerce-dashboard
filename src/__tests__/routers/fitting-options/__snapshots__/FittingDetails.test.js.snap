// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FittingDetails component Render the FittingDetails component with default props 1`] = `
<div>
  <div
    class="fitting-options__wrapper fitting-options__fitting-details"
  >
    <div
      class="fitting-options__selected"
      style="display: none;"
    >
      <div
        class="bookingItem bookingItem__fitting-kit"
        role="presentation"
      >
        <div>
          <span
            class="bookingItem__icon"
          />
        </div>
        <p
          class="bookingItem__title"
        >
          I would like to fit myself using Hemster's Fitting Kit
        </p>
      </div>
    </div>
    <div
      style="display: none;"
    >
      <div
        class="bookingItem bookingItem__fit-stylist bookingItem__small"
        role="presentation"
      >
        <div>
          <span
            class="bookingItem__icon"
          />
        </div>
        <p
          class="bookingItem__title"
        >
          I would like to book an appointment to meet with a Fit Stylist virtually
        </p>
      </div>
      <div
        class="bookingItem bookingItem__in-home-stylist bookingItem__small"
        role="presentation"
      >
        <div>
          <span
            class="bookingItem__icon"
          />
        </div>
        <p
          class="bookingItem__title"
        >
          I would like to book an In-Home Appointment with a Fit Stylist
        </p>
      </div>
    </div>
    <div
      class="indicator"
    >
      <span
        class="indicator__img"
      />
    </div>
  </div>
</div>
`;
